### PR TITLE
[EWS V2] Fix unexpected limit argument behavior of ews-search-mailboxes command

### PR DESCRIPTION
**Pull request by samuelFain**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [XSUP-42811](https://jira-dc.paloaltonetworks.com/browse/XSUP-42811)

## Description
Fixed an issue in ***ews-search-mailboxes*** command where the *limit* argument was capped at 100 results.

### CODE REVIEW

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml [hunk 1]
```diff
@@ -246,7 +246,7 @@ script:
       isArray: true
       name: mailbox-search-scope
     - defaultValue: '250'
-      description: Maximum number of results to return.
+      description: Maximum number of results to return. Default is 250.
```
Comment (samuelFain) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml:249]:
```suggestion
      description: Maximum number of results to return. Default is 100.
```

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 2]
```diff
@@ -805,8 +811,26 @@ def get_searchable_mailboxes(protocol):  # pragma: no cover
 
 
 def search_mailboxes(protocol, filter, limit=100, mailbox_search_scope=None, email_addresses=None):  # pragma: no cover
+    """
+    Search mailboxes for items matching the given filter.
+
+    Args:
+        protocol (Protocol): The EWS protocol object.
+        filter (str): The search filter to apply.
+        limit (int): The maximum number of results to return. Default value is SEARCH_MAILBOXES_LIMIT.
```
Comment (samuelFain) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:820]:
```suggestion
        limit (int): The maximum number of results to return. Default value is 100.
```

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml [hunk 3]
```diff
@@ -246,7 +246,7 @@ script:
       isArray: true
       name: mailbox-search-scope
     - defaultValue: '250'
```
Comment (michal-dagan) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml:248]:
```suggestion
    - defaultValue: '100'
```

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 4]
```diff
@@ -805,8 +811,26 @@ def get_searchable_mailboxes(protocol):  # pragma: no cover
...
+    Raises:
+        Exception: If both mailbox_search_scope and email_addresses are provided, or if no searchable mailboxes are found.
+    """
     mailbox_ids = []
-    limit = int(limit)
+    limit_argument = arg_to_number(limit)
+    if not limit_argument:
+        raise (DemistoException(f"Invalid limit value: {limit}. Please provide a valid integer."))
     if mailbox_search_scope is not None and email_addresses is not None:
         raise Exception("Use one of the arguments - mailbox-search-scope or email-addresses, not both")
```
Comment (omerKarkKatz) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:835]:
```suggestion
    if not limit_argument:
        raise (DemistoException(f"Invalid limit value: {limit}. Please provide a valid integer."))
        
    if mailbox_search_scope is not None and email_addresses is not None:
        raise Exception("Use one of the arguments - mailbox-search-scope or email-addresses, not both")
```

Comment (samuelFain) [reply Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py]: @omerKarkKatz I removed the redundant brackets around the `DemistoException` if that was this comment's intent, if there is something else, let me know :)

**Review by omerKarkKatz — APPROVED** [review]

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 5]
```diff
@@ -805,8 +811,26 @@ def get_searchable_mailboxes(protocol):  # pragma: no cover
...
+    Raises:
+        Exception: If both mailbox_search_scope and email_addresses are provided, or if no searchable mailboxes are found.
+    """
     mailbox_ids = []
-    limit = int(limit)
+    limit_argument = arg_to_number(limit)
+    if not limit_argument:
+        raise DemistoException(f"Invalid limit value: {limit}. Please provide a valid integer.")
     if mailbox_search_scope is not None and email_addresses is not None:
         raise Exception("Use one of the arguments - mailbox-search-scope or email-addresses, not both")
```
Comment (samuelFain) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:835]:
```suggestion
    if not limit_argument:
        raise DemistoException(f"Invalid limit value: {limit}. Please provide a valid integer.")

    if mailbox_search_scope is not None and email_addresses is not None:
        raise Exception("Use one of the arguments - mailbox-search-scope or email-addresses, not both")
```

**Comment by samuelFain** [comment]
@DeanArbel Need a force merge due to TPBs failures that are known issues (none of them related to the changes in this PR).

**Pull request merged**